From 57c6ceb445f3d125664a6e3aac79903c7332b9ad Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sat, 02 Jul 2022 00:50:14 +0300
Subject: [PATCH] [SW-93545] gaudi2: erom: fix tpm stuck after hard reset

remove pll init which isn't required by erom tpm and make sure
ARCs are privileged after LKD has unloader and done the hard reset.

Change-Id: I7dc2268ab7b35e5d7ac7ded70f3736a8868a97f1
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 1145dfc..9afd66e 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -115,6 +115,7 @@
 	$(PREBOOT_ROOT_DIR)/lib/common/lib_common.o \
 	$(HL_DEV_ARCH_COMMON)/tpm2.o \
 	$(HL_DEV_ARCH_COMMON)/tpm_auth.o \
+	$(HL_DEV_ARCH_COMMON)/security.o \
 	$(HL_DEV_ARCH)/../common/$(HL_TARGET).o \
 	$(HL_DEV_ARCH)/../common/common.o \
 	$(HL_DEV_ARCH)/../common/pll.o \
diff --git a/app/ext_rom/erom_tpm.c b/app/ext_rom/erom_tpm.c
index bb1e4b7..3a20a66 100644
--- a/app/ext_rom/erom_tpm.c
+++ b/app/ext_rom/erom_tpm.c
@@ -123,14 +123,19 @@
 	device_init_efuse();
 }
 
+__weak void ppboot_secure_arc(void) {}
+
 void erom_tpm(void)
 {
 	uint32_t offset = 0;
 	uint16_t size = 0;
 
 	device_init();
+
+	/* Set ARCs to privileged mode */
+	ppboot_secure_arc();
+
 	udelay_init();
-	hl_dev_pll_init();
 	hl_uart_init();
 	cpu_boot_if_init();
 
